assets/src/components/activities/oli_embedded/OliEmbeddedDelivery.tsx

```diff
@@ -0,0 +1,101 @@
+import { configureStore } from 'state/store';
+import React, { useEffect, useState } from 'react';
+import { OliEmbeddedModelSchema } from 'components/activities/oli_embedded/schema';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import * as ActivityTypes from 'components/activities/types';
+import {
+  DeliveryElement,
+  DeliveryElementProps,
+  DeliveryElementProvider,
+  useDeliveryElementContext,
+} from 'components/activities/DeliveryElement';
+import { activityDeliverySlice } from 'data/activities/DeliveryState';
+
+interface Context {
+  attempt_guid: string;
+  src_url: string;
+  activity_type: string;
+  server_url: string;
+  user_guid: string;
+  mode: string;
+  part_ids: string;
+}
+
+const EmbeddedDelivery = (props: DeliveryElementProps<OliEmbeddedModelSchema>) => {
+  const { state: activityState } = useDeliveryElementContext<OliEmbeddedModelSchema>();
+
+  const [context, setContext] = useState<Context>();
+  const [preview, setPreview] = useState<boolean>(false);
+
+  useEffect(() => {
+    fetchContext();
+  }, []);
+
+  const fetchContext = () => {
+    fetch('/jcourse/superactivity/context/' + activityState.attemptGuid, {
+      method: 'GET',
+    })
+      .then((response) => response.json())
+      .then((json) => {
+        setContext(json);
+      })
+      .catch((error) => {
+        // :TODO: display error somehow
+        setPreview(true);
+      });
+  };
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  window.adjustIframeHeight = (i, f) => {
```

Comment:
seems a bit odd to add something to the global space here

Reply:
Maintained for backward compatiblity. Some legacy client side still assuming this is available